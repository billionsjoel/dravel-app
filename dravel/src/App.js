import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <h1>dev ds dqsd </h1>
    </div>
  );
}

export default App;
