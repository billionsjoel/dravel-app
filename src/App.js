import './App.css';
import {
  Route,
  Routes,
  BrowserRouter,
  Navigate,
} from 'react-router-dom';
import { useSelector } from 'react-redux';
import Splash from './pages/splash';
import HomePage from './pages/homePage';
import DetailsPage from './pages/detailsPage';
import AddPage from './pages/addPage';
import DeletePage from './pages/deletePage';
import ReservationPage from './pages/reservation';
import MyReservation from './pages/myReservation';
import Reserve from './pages/reserve';
/* eslint-disable */

function App() {
  const isUser = useSelector((state) => state.dravelReducer.user);
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route
            path="*"
            element={<Navigate to="/" />}
          />
          {
            !isUser.length > 0 ?

            <Route path="/" element={<Splash />} />

              :
              <>
              <Route path="/" element={<HomePage />} />
              <Route path="/details" element={<DetailsPage />} />
              <Route path="/reservation" element={<ReservationPage />} />
              <Route path="/myreservation" element={<MyReservation />} />
              <Route path="/add" element={<AddPage />} />
              <Route path="/delete" element={<DeletePage />} />
              <Route path="/reserve" element={<Reserve />} />
              </>
        }
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
