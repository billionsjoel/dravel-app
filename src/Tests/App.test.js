import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

describe('It should test UI DOM components', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);

    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
