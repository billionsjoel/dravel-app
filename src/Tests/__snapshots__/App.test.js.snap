// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should test UI DOM components should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="cover"
    >
      <div
        class="nav"
      >
        <div
          class="logo"
        >
          <span
            class="fa fa-bars"
          >
             
          </span>
          <span>
            Dravel
          </span>
        </div>
        <div
          class="search"
        >
          <i
            aria-hidden="true"
            class="fa fa-search"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class="heading"
        >
          Dravel
        </div>
        <div
          class="cta"
        >
          <span
            aria-hidden="true"
            class="fa fa-cog p-2"
          />
          <div>
            login
          </div>
        </div>
      </div>
      <div
        class="footer fixed-bottom mb-4"
      >
        Â© Copyright Dravel 2022
      </div>
    </div>
  </div>
</DocumentFragment>
`;
